### PR TITLE
doc about how to compile with 3D, speedup build time with ninja

### PR DESCRIPTION
**Pull request by Gustry**
Some notes I found on the QGIS Dev ML from @m-kuhn and this morning with @nirvn about 3D.

Review welcome, cmake is not my cup of tee ;-)

Thanks for your tips.

### CODE REVIEW

File: doc/linux.t2t [hunk 1]
```diff
@@ -210,10 +210,39 @@ Build and install with ninja:
...
 ```
 ninja qgis
 ninja pycore
+# if it's on desktop related code only:
+ninja qgis_desktop
+```
+
+== Compiling with 3D ==
+
+QGIS 3D requires Qt53DExtras. Unfortunately, these headers have been removed from Qt upstream.
```
Comment (nyalldawson) [inline doc/linux.t2t]: This isn't needed everywhere -- only on debian based systems

Comment (3nids) [reply doc/linux.t2t]: MacOS too

**Comment by Gustry** [comment]
Can someone merge?

File: doc/linux.t2t [hunk 2]
```diff
@@ -210,10 +210,40 @@ Build and install with ninja:
...
 ```
 ninja qgis
 ninja pycore
+# if it's on desktop related code only:
+ninja qgis_desktop
+```
+
+== Compiling with 3D ==
+
+QGIS 3D requires Qt53DExtras. Unfortunately, these headers have been removed
```
Comment (m-kuhn) [inline doc/linux.t2t]: Can you split generic and debian specific instructions?

File: doc/linux.t2t [hunk 3]
```diff
@@ -210,10 +210,40 @@ Build and install with ninja:
...
+== Building different branches ==
+
+By using ``git worktree``, you can switch between different branches to use
+several sources in parallel, based on the same Git configuration.
+We recommand you to read the documentation about this Git command:
+```
+git commit
+git worktree add ../my_new_functionality
+cd ../my_new_functionality
+git rebase -i qgis/master
```
Comment (m-kuhn) [inline doc/linux.t2t]:
```suggestion
git fetch qgis/master
git rebase -i qgis/master
```

This will be useful as well.

Comment (Gustry) [reply doc/linux.t2t]: Done

File: doc/linux.t2t (hunk 2 above)
Comment (Gustry) [reply doc/linux.t2t]: To be honest, I'm not sure which ones are for Debian only or other platform. This was just my feedback from Ubuntu.

Comment (Gustry) [reply doc/linux.t2t]: Gentle ping @m-kuhn during this remote code sprint ;-) 
I compile QGIS only on Ubuntu.

Comment (m-kuhn) [reply doc/linux.t2t]: WITH_3D is generic.
Everything else is Debian specific. Just make a subsection with debian in the title.
And I would remove the term "Unfotunately", I don't think this adds much value for the reader.

Comment (Gustry) [reply doc/linux.t2t]: Done @m-kuhn

**Review by m-kuhn — APPROVED** [review]

**Comment by Gustry** [comment]
Thanks @rouault !

**Pull request merged**